pwndbg/arguments.py

```diff
@@ -196,6 +205,72 @@ def arguments(abi=None):
         yield argname(i, abi), argument(i, abi)
 
 
+def enhance_printf_args(orig_args, format_str):
+    args = orig_args.copy()
+
+    vararg_arg = args.pop()
+    if vararg_arg.name != 'vararg':
+        return orig_args
+
+    # NOTE: this table is currently not quite used
+    len_modifier_table = {
+        'hh': 'char',
+        'll': 'long long',
+        'l': 'long',
+        'j': 'intmax_t',
+        'z': 'size_t',
+        't': 'ptrdiff_t',
+        'L': 'long double'
+    }
+
+    i = 0
+    len_format_str = len(format_str)
+
+    while i < len_format_str:
+        if format_str[i] != '%':
+            i += 1
+            continue
+
+        if i + 1 < len_format_str and format_str[i + 1] == '%':
+            i += 2
+            continue
+        fmt_specifier_start = i
+        i += 1
+
+        # FIXME: should recognize '*' and accept additional int arguments
+        while i < len_format_str and (format_str[i] in '-+ #0*.' or format_str[i].isdigit()):
+            i += 1
+
+        len_modifier_2 = format_str[i:i+2]
+
+        specifier_type = None
+        if len_modifier_2 in ('hh', 'll'):
+            specifier_type = len_modifier_table[len_modifier_2]
+            i += 2
+        else:
+            len_modifier_1 = format_str[i]
+            if len_modifier_1 in len_modifier_table:
+                specifier_type = len_modifier_table[len_modifier_1]
+                i += 1
+
+        conv_specifier = format_str[i]
```

Comment:
This may actually raise `IndexError` if the format string is malformed.

Same for this line:

https://github.com/pwndbg/pwndbg/blob/6cbd6a177ff343ca317928ee792b240f7e1abf55/pwndbg/arguments.py#L251